python/cudf/cudf/core/dataframe.py

```diff
@@ -6081,6 +6082,42 @@ def nunique(self, axis=0, dropna=True):
 
         return cudf.Series(super().nunique(method="sort", dropna=dropna))
 
+    def _sample_axis_1(
+        self,
+        n: int,
+        weights: Optional[ColumnLike],
+        replace: bool,
+        random_state: np.random.RandomState,
+        ignore_index: bool,
+    ):
+        if replace:
+            raise NotImplementedError(
+                "Sample is not supported for axis 1/'columns' "
+                "when 'replace=True'"
+            )
+
+        if n > 0 and self.shape[1] == 0:
+            raise ValueError(
+                "Cannot take a sample larger than 0 when axis is empty"
+            )
+
+        columns = np.asarray(self._data.names)
+        if not replace and n > columns.size:
```

Comment:
No need to do this cast explicitly, rely on the library to handle it for you. Especially important if we start adding support for other random state objects, it's easier to pass a common object like a tuple that everything should be able to handle.
```suggestion
        columns = self._data.names
        if not replace and n > len(columns):
```